service/common/src/main/java/org/apache/polaris/service/events/listeners/FileBufferPolarisEventListener.java

```diff
@@ -0,0 +1,150 @@
+        }
+        try {
+            ObjectOutputStream oos = bufferShard.oos;
+            oos.writeObject(polarisEvent);
+            oos.flush();
+            oos.reset();
+        } catch (IOException e) {
+            LOGGER.error("Unable to write event: {} to buffer shard: {}", polarisEvent, e);
+        }
+
+        if (ingestionTime - bufferShard.polarisEvent.getTimestampMs() > this.timeToFlush) {
+            // Create a new shard (file and writer)
+            buffers.put(shardNum, createShard(shardNum));
+            // Create and fire task to persist the events in the old shard file
+            PolarisMetaStoreManager polarisMetaStoreManager = metaStoreManagerFactory.getOrCreateMetaStoreManager(callCtx.getRealmContext());
+            PolarisBaseEntity taskEntity =
+                    new PolarisEntity.Builder()
+                            .setId(polarisMetaStoreManager.generateNewEntityId(callCtx.getPolarisCallContext()).getId())
+                            .setCatalogId(0L)
+                            .setName("persist_logs_and_cleanup_" + bufferShard.shardFile.getName())
+                            .setType(PolarisEntityType.TASK)
+                            .setSubType(PolarisEntitySubType.NULL_SUBTYPE)
+                            .setCreateTimestamp(System.currentTimeMillis())
+                            .build();
+            Map<String, String> properties = new HashMap<>();
+            properties.put(
+                    PolarisTaskConstants.TASK_TYPE,
+                    String.valueOf(AsyncTaskType.LOG_PERSISTENCE_AND_CLEANUP.typeCode()));
+            properties.put("file", bufferShard.shardFile.getAbsolutePath());
+            taskEntity.setPropertiesAsMap(properties);
+            EntityResult entityResult = polarisMetaStoreManager.createEntityIfNotExists(callCtx.getPolarisCallContext(), null, taskEntity);
+            taskExecutor.addTaskHandlerContext(entityResult.getEntity().getId(), callCtx);
+        }
+    }
+
+    private BufferShard createShard(int shardNum) {
+        File file = new File(System.getProperty("java.io.tmpdir") + "/polaris-event-buffer-shard-" + shardNum + "file_timestamp-" + System.currentTimeMillis());
+        try {
+            file.createNewFile();
+            ObjectOutputStream oos = new ObjectOutputStream(new FileOutputStream(file, true));
+            return new BufferShard(null, file, oos);
+        } catch (IOException e) {
+            LOGGER.error("Buffer shard {} was unable to initialize: {}", shardNum, e);
+        }
+        return null;
+    }
+
+    @SuppressWarnings("ClassCanBeStatic")
+    private class BufferShard {
+        public PolarisEvent polarisEvent;
+        public File shardFile;
+        public ObjectOutputStream oos;
+
+        public BufferShard(PolarisEvent polarisEvent, File shardFile, ObjectOutputStream oos) {
+            this.polarisEvent = polarisEvent;
+            this.shardFile = shardFile;
+            this.oos = oos;
```

Comment:
Feels like maybe the management of the filehandle and the stream can be delegated to this class